api/src/main/java/org/openmrs/module/registrationcore/api/mpi/common/MpiPatient.java

```diff
@@ -0,0 +1,20 @@
+package org.openmrs.module.registrationcore.api.mpi.common;
+
+import org.openmrs.Patient;
+
+/**
+ * Defines a MpiPatient in the system. A mpi patient is simply an extension of a patient and all that that
+ * implies.
+ */
+public class MpiPatient extends Patient {
```

Comment:
What I'm saying is that you have a compound object that has a patient as a property. You can use that property as a common patient.

E.g. we'd introduce something like

    class MatchingPatientResult {
        // never null
        Patient patient; 
        // if non-null, this is an MPI patient.
        Object rawMpiResponse;
    }

Then in any code you can do "matchingPatientResults.patient" to refer to the patient.

Something like this principle: https://en.wikipedia.org/wiki/Composition_over_inheritance

Also, the fact that we have to introduce a fake hibernate mapping xml to support this is really incorrect.